jme3-android/src/main/java/com/jme3/input/android/AndroidJoyInput.java

```diff
@@ -215,7 +215,7 @@ public Joystick[] loadJoysticks(InputManager inputManager) {
         if (!disableSensors) {
             joystickList.add(sensorJoyInput.loadJoystick(joystickList.size(), inputManager));
         }
-        return joystickList.toArray( new Joystick[joystickList.size()] );
```

Comment:
Why is this considered more performant than the old code? Passing a zero-length array results in 2 heap allocations, only one of which is necessary. The same question also applies to many of the other changes in this PR.

Reply:
This is why https://shipilev.net/blog/2016/arrays-wisdom-ancients/

Reply:
You can skip to the conclusion if you are not interested in the background.  Bottom line is that for counter-intuitive reasons, toArray(new Foo[0]) is faster than toArray(new Foo[specificSize]).  
I read this lore correction some years back.

Reply:
Facinating.